src/Features/CSharp/Portable/ExtractMethod/CSharpMethodExtractor.Analyzer.cs

```diff
@@ -129,6 +130,72 @@ protected override bool ReadOnlyFieldAllowed()
                 var scope = this.SelectionResult.GetContainingScopeOf<ConstructorDeclarationSyntax>();
                 return scope == null;
             }
+
+            protected override ITypeSymbol GetSymbolType(SemanticModel semanticModel, ISymbol symbol)
+            {
+                var selectionOperation = semanticModel.GetOperation(this.SelectionResult.GetContainingScope());
+
+                switch (symbol)
+                {
+                    case ILocalSymbol local:
```

Comment:
Can we just unify the ILocalSymbol and IParameterSymbol branches? Consider something like:

```
ILocalSymbol? localSymbol = symbol as ILocalSymbol;
IParameterSymbol? parameterSymbol = symbol as IParameterSymbol;

// if both null, bail

var references = selectionOperation.DescendantsAndSelf()
                                .Where(o => o switch
                                {
                                    ILocalReferenceOperation localReference => localReference.Local == localSymbol,
                                    IParameterReferenceOperation parameterReference => parameterReference.Parameter == parameterSymbol,
```

The idea being one is always null, and the DescendandsAndSelf will match a bit more but it'll just not match.